ignite/engines/evaluator.py

```diff
@@ -42,7 +42,7 @@ def _inference(batch):
         model.eval()
         x, y = _prepare_batch(batch)
         y_pred = model(x)
-        return y_pred.data.cpu(), y.data.cpu()
+        return to_tensor(y_pred, cpu=not cuda), to_tensor(y, cpu=not cuda)
```

Comment:
nit: cpu could always be true here. I think its a no-op if it's already on the cpu

Reply:
Do you think it would not impact metrics computation time if it is on cpu ?